nameserver/nameserver.go

```diff
@@ -137,6 +138,19 @@ func (n *Nameserver) ContainerDied(ident string) {
 	n.broadcastEntries(entries...)
 }
 
+func (n *Nameserver) ContainerDisconnected(ident string) {
+	n.Lock()
+	entries := n.entries.tombstone(n.ourName, func(e *Entry) bool {
+		if e.ContainerID == ident {
+			n.infof("container %s disconnected; tombstoning entry %s", ident, e.String())
```

Comment:
It can be connected to multiple networks, of which several can map to the same plugin.

Without having thought about it deeply, I suspect the right thing to do is remove the specific address in `ContainerDisconnected` in the plugin, and not handle `ContainerDisconnected` here.